fastly/platform_tls.go

```diff
@@ -0,0 +1,390 @@
+package fastly
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/google/jsonapi"
+)
+
+// GetPlatformPrivateKeysInput is an input to the GetPlatformPrivateKeys function.
+// Allowed values for the fields are described at https://docs.fastly.com/api/platform-tls
+type GetPlatformPrivateKeyInput struct {
+	ID string
+}
+
+// PlatformPrivateKeysResponse is a response from the service tls API endpoint
+type PlatformPrivateKeysResponse struct {
+	Status  string                `mapstructure:"status"`
+	Meta    map[string]string     `mapstructure:"meta"`
+	Message string                `mapstructure:"msg"`
+	Data    []*PlatformPrivateKey `mapstructure:"data"`
+}
+
+// PlatformPrivateKeyResponse is a response from the service tls API endpoint
+type PlatformPrivateKeyResponse struct {
+	Status  string              `mapstructure:"status"`
+	Meta    map[string]string   `mapstructure:"meta"`
+	Message string              `mapstructure:"msg"`
+	Data    *PlatformPrivateKey `mapstructure:"data"`
+}
+
+// PlatformPrivateKey .
+type PlatformPrivateKey struct {
+	ID            string                        `mapstructure:"id"`
+	Type          string                        `mapstructure:"type"`
+	Attributes    PlatformPrivateKeysAttributes `mapstructure:"attributes"`
+	Comment       string                        `mapstructure:"comment"`
+	CustomerID    string                        `mapstructure:"customer_id"`
+	CreatedAt     *time.Time                    `mapstructure:"created_at"`
+	UpdatedAt     *time.Time                    `mapstructure:"updated_at"`
+	DeletedAt     *time.Time                    `mapstructure:"deleted_at"`
+	ActiveVersion uint                          `mapstructure:"version"`
+	Versions      []*Version                    `mapstructure:"versions"`
+}
+
+// PlatformPrivateKeysAttributes .
+type PlatformPrivateKeysAttributes struct {
+	KeyLength     string     `mapstructure:"key_length"`
+	KeyType       string     `mapstructure:"key_type"`
+	Name          string     `mapstructure:"name"`
+	CreatedAt     *time.Time `mapstructure:"created_at"`
+	Replace       bool       `mapstructure:"replace"`
+	PublicKeySHA1 string     `mapstructure:"public_key_sha1"`
+}
+
+// GetPlatformPrivateKeys returns stats data based on GetPlatformPrivateKeysInput
+func (c *Client) GetPlatformPrivateKeys() (*PlatformPrivateKeysResponse, error) {
+
+	p := "/tls/private_keys"
+
+	r, err := c.Get(p, &RequestOptions{
+		Params: map[string]string{},
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	var ppkr *PlatformPrivateKeysResponse
+	if err := decodeJSON(&ppkr, r.Body); err != nil {
+		return nil, err
+	}
```

Comment:
As per comments above, this is a JSONAPI response and therefore we should unmarshal using the [`jsonapi`](https://github.com/google/jsonapi) package so you don't need to model the response envelope.  The library provides two helpers, the singular `UnmarshalPayload` and `UnmarshalManyPayload` for multiple. We have to use reflection for the many payload to let jsonapi know the type its decoding into. More info here: https://github.com/google/jsonapi#unmarshalmanypayload

```suggestion
	data, err := jsonapi.UnmarshalManyPayload(resp.Body, reflect.TypeOf(new(PlatformPrivateKey)))
	if err != nil {
		return nil, err
	}
```